app/models/user.rb

```diff
@@ -1,4 +1,29 @@
 class User < ActiveRecord::Base
   has_many :bids
   validates :email, email: true, allow_blank: true
+  before_save :clear_sam_status_if_duns_changed
```

Comment:
I don't think this belongs inside User. Also it is specific to vendor users, not admins. All this is a wrapper around user data that adds to it.